sirepo/pkcli/job_cmd.py

```diff
@@ -80,6 +80,8 @@ def _background_percent_complete(msg, template, is_running):
 
 def _dispatch_compute(msg, template):
     def _op(expect_file):
+        if "dataFileUri" in msg:
```

Comment:
This is where things get messy with interfaces. If `msg.data` were of a specific type, then it would need to be there when originally called. I suggest `<cmd>_dispatch` take an extra arg data_file_uri (or dataFileUri) which may be None, i.e. the call would look like:

`r = getattr(template_common, f"{msg.jobCmd}_dispatch")(msg.data, data_file_uri=msg.get('dataFileUri'))`

These would cascade to all stateful_compute and stateless_compute calls so that "data" can be pure. There's precedence for the rpn functions which are dispatched and pass along values (not thrilled with this code but it's basically the same idea).

The updates to all the stateless and stateful computes are easy and only the template_common code needs a bit of special handling (and probably should just cascade `**kwargs` which would simplify the code anyway).